vortex-dtype/src/dtype.rs

```diff
@@ -11,7 +12,7 @@ use crate::nullability::Nullability;
 use crate::{ExtDType, PType};
 
 /// A name for a field in a struct
-pub type FieldName = Arc<str>;
```

Comment:
So `Arc<str>` has 24 bytes overhead (usize ptr, usize len, usize refcount), if I'm following inline-array docs then InlineStr should have what, 2 bytes of overhead in the worst cast?

Reply:
I think it can be worse if we clone a ton, but realistically yes.